amalgamation/Makefile

```diff
@@ -28,21 +33,19 @@ nnvm.d:
 	./prep_nnvm.sh
 
 dmlc.d: dmlc-minimum0.cc
-	${CXX} ${CFLAGS} -MD -MF $@ \
+	${CXX} ${CFLAGS} -M -MT dmlc-minimum0.o \
 	-I ${MXNET_ROOT}/dmlc-core/include \
-	-D__MIN__=$(MIN) -c $+
-	rm dmlc-minimum0.o
+	-D__MIN__=$(MIN) $+ > dmlc.d
 
 
 mxnet_predict0.d: mxnet_predict0.cc nnvm.d dmlc.d
-	${CXX} ${CFLAGS} -MD -MF $@ \
+	${CXX} ${CFLAGS} -M -MT mxnet_predict0.o \
```

Comment:
MT expect the literal target name while MF will use the rule to determine. If using MF here, mxnet_predict0.d will be used as target in the .d file which is not expected.

Original build script compile the input so that .o will be used. but it is not needed. So I change the rule to only preprocess file.